docs/api/ref/PrivateChannel.md

```diff
@@ -75,88 +87,107 @@ The 'client' application retrieves a `Channel` by raising an intent with context
+    (event: PrivateChannelAddContextListenerEvent) => { ... }
+);
+```
 
 **See also:**
 
-- [`Channel.addContextListener`](Channel#addcontextlistener)
+- [`PrivateChannelEventTypes`](./Events#privatechanneleventtypes)
+- [`PrivateChannelEvent`](./Events#privatechannelevent)
+- [`EventHandler`](./Events#eventhandler)
 
-### `onUnsubscribe`
+### `disconnect`
 
 ```ts
-onUnsubscribe(handler: (contextType?: string) => void): Listener;
+disconnect(): Promise<void>;
 ```
 
-Adds a listener that will be called whenever the remote app invokes `Listener.unsubscribe()` on a context listener that it previously added.
+May be called to indicate that a participant will no longer interact with this channel.
 
-Desktop Agents MUST call this when disconnect() is called by the other party, for each listener that they had added.
+After this function has been called, Desktop Agents SHOULD prevent apps from broadcasting on this channel and MUST automatically call Listener.unsubscribe() for each listener that they've added (causing any event handler for `unsubscribe` events added by the other party to be called) before triggering any handlers for `disconnect` events added by the other party.
 
 **See also:**
 
 - [`Listener`](Types#listener)
 
-### `onDisconnect`
+## Deprecated Functions
+
+### `onAddContextListener`
 
 ```ts
-onDisconnect(handler: () => void): Listener;
+onAddContextListener(handler: (contextType?: string) => void): Listener;
 ```
 
-Adds a listener that will be called when the remote app terminates, for example when its window is closed or because disconnect was called. This is in addition to calls that will be made to onUnsubscribe listeners.
-
-**See also:**
+Deprecated in favour of the async `addEventListener("addContextListener", handler)` function.
 
-- [`disconnect`](#disconnect)
+Adds a listener that will be called each time that the remote app invokes addContextListener on this channel.
 
-### `disconnect`
+### `onUnsubscribe`
 
 ```ts
-disconnect(): void;
+onUnsubscribe(handler: (contextType?: string) => void): Listener;
 ```
 
-May be called to indicate that a participant will no longer interact with this channel.
+Deprecated in favour of the async `addEventListener("unsubscribe", handler)` function.
 
-After this function has been called, Desktop Agents SHOULD prevent apps from broadcasting on this channel and MUST automatically call Listener.unsubscribe() for each listener that they've added (causing any `onUnsubscribe` handler added by the other party to be called) before triggering any onDisconnect handler added by the other party.
+Adds a listener that will be called whenever the remote app invokes `Listener.unsubscribe()` on a context listener that it previously added.
 
-**See also:**
+### `onDisconnect`
 
-- [`onUnsubscribe`](#onunsubscribe)
-- [`Listener`](Types#listener)
+```ts
+onDisconnect(handler: () => void): Listener;
+```
+
+Deprecated in favour of the aysnc `addEventListener("disconnect", handler)` function.
```

Comment:
```suggestion
Deprecated in favour of the async `addEventListener("disconnect", handler)` function.
```

Reply:
Fixed, @hughtroeger or @bingenito pr @finos/fdc3-maintainers can you re-approve so I can merge pls? (There was a push so past reviews are invalidated)